### DISCUSSION TITLE
Is scanning hosts by URL a targeted use case?

### DISCUSSION THREAD

**Discussion by timlst** [zmap/zmap#889] [Q&A] [answered]
Hi,

[motivation for my question]
in a current project, we're trying to use the combination of `zmap`+`zgrab` to scan a list of URLs (e.g. Tranco 1M).
For us it makes a lot of sense to pre-filter using `zmap` to timing out with `zgrab`.
However, we absolutely need  (IP, hostname) for these which has led to a weird pipeline of resolving addresses first, throwing them into `zmap` and then recombining them.

[actual question]
Is the mass scanning of hosts (by DNS name) a targeted use case? Although the command line supports host names, it seems like an afterthought: for example there is no `--list-of-hostname-file` parameter and the output does not allow a `hostname` field or anything.
I was considering opening an issue, but wanted to gauge first, whether this is even considered in-scope :)

Thanks and best regards!

**Comment by timlst** [comment]
I believe my question has partially resolved itself after snooping around: I just found out `--allowlist-file` allows me to supply hostnames as well (which is not documented afaict). Therefore my question boils down to:

Would a `hostname` output field be considered out-of-scope?

**Comment by zakird** [comment, marked as answer]
There isn't a _great_ way to do this, because ZMap needs to know at the start of the scan what IPs that it has to scan data-structure-wise. In the past when we had to do this, what we did was... 

1) Run names through ZDNS and get IPs
2) ZMap IPs
3) Join back together, and feed back ZGrab as (name,ip)
